### PR TITLE
refactor: move ReaderTaskEither from examples into src

### PR DESCRIPTION
**Pull request by leemhenson**
see https://github.com/gcanti/fp-ts/issues/446

### CODE REVIEW

File: test/ReaderTaskEither.ts [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+  it('of', () => {
+    return readerTaskEither
+      .of(1)
+      .run({})
+      .run()
+      .then(e => assert.deepEqual(e, eitherRight(1)))
+  })
+
+  it('of', () => {
```
[the remark this answers is not in the document]
Comment (leemhenson) [reply test/ReaderTaskEither.ts]: If I drop `of` then `readerTaskEither` no longer satisfies `Monad3<URI>`.

[the remark this answers is not in the document]
Comment (gcanti) [reply test/ReaderTaskEither.ts]: I mean the `of` **method**, There's still the `of` static function which is member of `Monad3`

[the remark this answers is not in the document]
Comment (leemhenson) [reply test/ReaderTaskEither.ts]: Oh I see what you mean. Done.

**Comment by gcanti** [comment]
Thanks @leemhenson

**Pull request merged**